python/cugraph-dgl/cugraph_dgl/dataloading/utils/sampling_helpers.py

```diff
@@ -29,53 +29,114 @@ def cast_to_tensor(ser: cudf.Series):
     return torch.as_tensor(ser.values, device="cuda")
 
 
-def _get_tensor_ls_from_sampled_df(df):
+def _split_tensor(t, split_indices):
+    """
+    Split a tensor into a list of tensors based on split_indices.
+    """
+    # TODO: Switch to something below
+    # return [t[i:j] for i, j in zip(split_indices[:-1], split_indices[1:])]
+    if split_indices.device.type != "cpu":
+        split_indices = split_indices.to("cpu")
+    return torch.tensor_split(t, split_indices)
+
+
+def _get_id_tensor_boundaries(id_tensor):
+    """
+    Get the indices where the id_tensor changes values.
+    """
+    offset_indices = (id_tensor[1:] != id_tensor[:-1]).nonzero(as_tuple=True)[0] + 1
+    offset_indices = torch.cat(
+        [torch.tensor([0], device=id_tensor.device), offset_indices]
+    )
+    if id_tensor[-1] != id_tensor[-2]:
+        offset_indices = torch.cat(
+            [
+                offset_indices,
+                torch.tensor([len(id_tensor)], device=offset_indices.device),
+            ]
+        )
+    return offset_indices
+
+
+def _get_renumber_map(df):
+    map = df["map"]
+    df.drop(columns=["map"], inplace=True)
+
+    map_starting_offset = map.iloc[0]
+    renumber_map = map[map_starting_offset:].dropna().reset_index(drop=True)
+    offsets_per_batch = map[1:map_starting_offset].reset_index(drop=True)
```

Comment:
Is this correct?  You're separating the offsets from the map, and reindexing the map, but you're not subtracting off `map_starting_offset` from the elements in `offsets_per_batch`.

Reply:
I think you need to change this to something like:
```
offsets_per_batch = map[1:map_starting_offset].reset_index(drop=True) - map_starting_offset
```

Reply:
I think the variable name was wrong . Changed it to  `renumber_map_batch_indices ` .  

I think the implementation is correct but let me test more.

Reply:
````
map
4       # start of batch 0
10     # start of batch 1
15     # start of batch 2
17     # end of batch 2
6       # first renumber map starts here, at index 4
5
2
3
8
1
1       # second renumber map starts here, at index 10
8
2
11
14
8       # third renumber map starts here, at index 15
9
         # third renumber map ends here, at the length (17)
```

So, to get the map for each batch, you do:
```
for b in range(num_batches):
     indices_batch_b = map.iloc[[b, b+1]]
     start_ix_map = indices_batch_b.iloc[0]
     end_ix_map = indices_batch_b.iloc[1]
     map_batch_b = map.iloc[start_ix_map:end_ix_map]

     # do the rest of the processing for the current batch
```

Reply:
This seems to be resolved with your latest change 👍

Reply:
Thanks for catching this @alexbarghi-nv .  I appreciate the detailed explanation. 


I added a test to catch this behavior. 

https://github.com/rapidsai/cugraph/blob/4c2a42de21edb472156f96423dac51d8a1457e23/python/cugraph-dgl/tests/test_utils.py#L68-L75




@alexbarghi-nv , Can you verify that the map in below dataframe  is valid , please. 


https://github.com/rapidsai/cugraph/blob/4c2a42de21edb472156f96423dac51d8a1457e23/python/cugraph-dgl/tests/test_utils.py#L35-L43

Reply:
LGTM